assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -24,32 +24,45 @@ import { useEffect, useCallback } from 'react';
  * Internal dependencies
  */
 import { useAPI } from '../api';
+import { useConfig } from '../config';
 import useUploadVideoFrame from './utils/useUploadVideoFrame';
 import useMediaReducer from './useMediaReducer';
 import Context from './context';
 
 function MediaProvider({ children }) {
   const { state, actions } = useMediaReducer();
-  const { uploadVideoFrame } = useUploadVideoFrame();
-  const { mediaType, searchTerm } = state;
+
+  const { processing, media, mediaType, searchTerm } = state;
   const {
     fetchMediaStart,
     fetchMediaSuccess,
     fetchMediaError,
     resetFilters,
     setMediaType,
     setSearchTerm,
+    setProcessing,
+    removeProcessing,
+    updateMediaElement,
   } = actions;
-
+  const { uploadVideoFrame } = useUploadVideoFrame({
+    processing,
```

Comment:
Can `useUploadVIdeoFrame()` just resolve all of these dependencies inside via `useMediaReducer()`?